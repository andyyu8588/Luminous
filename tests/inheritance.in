class Yunze {
    public x;
    public ok;

    playGames() {
        this.x = "Yunze played some games.";
        print(this.x);
    }
    okkk() {
        this.ok = 1;
        print(this.ok);
        print(this.x);
    }
}

class Andy inherits Yunze {
    playGames() {
        this.x = "Andy played some games.";
        print(this.x);
    }
    yunzePlayGames() {
        super.playGames();
    }
}

andy = Andy();
andy.playGames();
andy.okkk();
print(andy.ok);

yunze = Yunze();
yunze.playGames();
yunze.okkk();
print(yunze.ok);

andy.yunzePlayGames();
print(andy.x);