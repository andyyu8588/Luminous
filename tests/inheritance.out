Andy played some games.
1
Andy played some games.
1
Yunze played some games.
1
Yunze played some games.
1
Yunze played some games.
Yunze played some games.
